### PR TITLE
Compatibility for compiling on Linux

### PR DESCRIPTION
**Pull request by jhscheer**
While tinkering with my warriors I changed a few things that were bugging me:

- I couldn't compile on my linux box
- I didn't like the focus switch to some browser window when starting
- there was a problem with the websockets when using localhost (I don't know the root cause for this)
- there was no message in case the warriors directory is missing
- there was no message in case only one warrior is found

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,17 @@
 VERSION=$(shell git tag|tail -n1)
 R2W=r2wars-$(VERSION)
 
+ifeq ($(OS),Windows_NT)
+    build := msbuild
```
Comment (radare) [inline Makefile]: Do not indent here

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+else
+    build := xbuild
+	endif
```
Comment (radare) [inline Makefile]: Wrong indent

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,17 @@
 VERSION=$(shell git tag|tail -n1)
 R2W=r2wars-$(VERSION)
 
+ifeq ($(OS),Windows_NT)
+    build := msbuild
+else
+    build := xbuild
```
Comment (radare) [inline Makefile]: Use uppercase for variablenames

File: Makefile (hunk 1 above)
Comment (radare) [reply Makefile]: Strange... msbuild works in mac too. Which version of mono are u using? Because im using mono on mac with msbuild

**Comment by radare** [comment]
The other changes lgtm

File: Makefile (hunk 1 above)
Comment (jhscheer) [reply Makefile]: I installed mono 5.14 from the packages provided by mono-project.com
Their packages (at least for linux, the others I don't know) do not contain msbuild.
According to https://lists.dot.net/pipermail/mono-list/2017-May/052476.html, msbuild is now a separate component and could be compiled from source from their github page.

**Comment by retauks** [comment]
Hi, i have a new version no pushed, i apply all changes to this version, tnx.

**Comment by radare** [comment]
ping

**Comment by trufae** [comment]
there are conflicts here, the pr must be rebased and fixed or closed. is this still an issue? r2wars is not working on linux?

**Comment by hmht** [comment]
For me r2wars worked great on linux with mono installed.

File: Program.cs [hunk 4]
```diff
@@ -26,9 +26,9 @@ static void Main(string[] args)
             {
                 var nancyHost = new Nancy.Hosting.Self.NancyHost(new Uri(httpUrl), new CustomBootstrapper());
                 nancyHost.Start();
-                Console.WriteLine("Web server running...");
+                Console.WriteLine("Web server running at " + httpUrl);
 
-                Process.Start(httpUrl);
+                /* Process.Start(httpUrl); */
```
Comment (trufae) [inline Program.cs]: Why commented?

File: Torneo.cs [hunk 5]
```diff
@@ -106,7 +106,7 @@ void runnextcombat(bool startcombat)
                     actualcombatscore[j].Score += new HighestPointsScore(0);
                     j++;
                 }
-                string tmp = string.Format("Iniciando combate {0} {1} vs {2}", ncombat + 1, actualcombatnames[0], actualcombatnames[1]);
+                string tmp = string.Format("Combat initialized {0} {1} vs {2}", ncombat + 1, actualcombatnames[0], actualcombatnames[1]);
```
Comment (trufae) [inline Torneo.cs]: I think “starting combat” is more correct in this context

**Review by trufae — CHANGES_REQUESTED** [review]
Apart from that lgtm. Can we merge?

File: Program.cs (hunk 4 above)
Comment (jhscheer) [reply Program.cs]: Why commented and not deleted?
I don't remember.
The line can be deleted.

**Comment by trufae** [comment]
there are conflicts, can  you please rebase?

**Comment by jhscheer** [comment]
> there are conflicts, can you please rebase?

yes, I'll do it tonight.

**Comment by jhscheer** [comment]
> there are conflicts here, the pr must be rebased and fixed or closed. is this still an issue? r2wars is not working on linux?

* It looks like most commits of this PR were "manually" added/merged to master in 2018 by @sanguinawer.
* I hope I solved the remaining conflicts.
* I also undid my changes in Makefile since calling msbuild depends on how mono/dotnet were installed.
i.e. with [mono-project.com](https://www.mono-project.com/download/stable/) you can run `msbuild` or `xbuild`, with [docs.microsoft.com/en-us/dotnet/core/install/linux](https://docs.microsoft.com/en-us/dotnet/core/install/linux) you need to run `dotnet build`.

**Comment by trufae** [comment]
Arg, seems like current code doesnt work on linux; could you update this pr and ill quickly merge? Thanks for your work! And sorry for not merging before. I had some build problems during r2con

**Comment by trufae** [comment]
Im closing this as long as current code build and runs on linux properly. we are now shipping a Docker image, and may be good to add a CI job for it and maybe an electron ui on top of that to make it one-click app :)

thanks! feel free to submit another PR with the changes in case i missed anything

**Pull request closed without merging**